File: db-services/db-user-service/src/main/java/com/qreal/wmp/db/user/model/diagram/Diagram.java
package com.qreal.wmp.db.user.model.diagram;

import com.qreal.wmp.thrift.gen.TDiagram;
import lombok.Data;

import java.io.Serializable;
import java.util.HashSet;
import java.util.Set;
import java.util.stream.Collectors;

/** Diagram (now only graphs).*/
@Data
public class Diagram implements Serializable {
    private Long id;

    private String name;

    private Set<DefaultDiagramNode> nodes = new HashSet<>();

    private Set<Link> links = new HashSet<>();

    public Diagram() {
    }

    /** Constructor-converter from Thrift TDiagram to Diagram.*/
    public Diagram(TDiagram tDiagram) {

        if (tDiagram.isSetId()) {
            id = tDiagram.getId();
        }

        if (tDiagram.isSetName()) {
            name = tDiagram.getName();
        }

        if (tDiagram.isSetNodes()) {
            nodes = tDiagram.getNodes().stream().map(DefaultDiagramNode::new).collect(Collectors.toSet());
        }

        if (tDiagram.isSetLinks()) {
            links = tDiagram.getLinks().stream().map(Link::new).collect(Collectors.toSet());
        }

    }

    /** Converter from Diagram to Thrift TDiagram.*/
    public TDiagram toTDiagram() {

        TDiagram tDiagram = new TDiagram();

        if (id != null) {
            tDiagram.setId(id);
        }

        if (name != null) {
            tDiagram.setName(name);
        }

        if (nodes != null) {
            tDiagram.setNodes(nodes.stream().map(DefaultDiagramNode::toTDefaultDiagramNode).
                    collect(Collectors.toSet()));
        }

        if (links != null) {
            tDiagram.setLinks(links.stream().map(Link::toTLink).collect(Collectors.toSet()));
        }

        return tDiagram;
    }
}

File: editor-service/src/main/webapp/app/common/menu/model/Diagram.ts
/// <reference path="../../../../resources/thrift/editor/EditorService_types.d.ts" />
/// <reference path="../../../../resources/thrift/struct/Diagram_types.d.ts" />
/// <reference path="../../../../resources/thrift/editor/EditorServiceThrift.d.ts" />
/// <reference path="../../../../resources/types/thrift/Thrift.d.ts" />

export class Diagram {

    private id: number;
    private name: string;

    public static createFromDAO(diagram: TDiagram) {
        return new Diagram(diagram.id, diagram.name)
    }

    constructor(id: number, name: string) {
        this.id = id;
        this.name = name;
    }

    public setId(id: number): void {
        this.id = id;
    }

    public getId(): number {
        return this.id;
    }

    public setName(name: string): void {
        this.name = name;
    }

    public getName(): string {
        return this.name;
    }

}